src/filter.jl

```diff
@@ -0,0 +1,27 @@
+"""
+Use Model `M` to track blurry light sources through
+frames in imgs. `X0` are the locations and `P0` 
+uncertainty of the locations. 
+`params` are arguments to `track`, eg. window-size.
+
+Returns matrix of the filtered states with a row per light sources
+and a column per frame.
+"""
+function trackandfilter(M, imgs, X0, P0, params...)
+    nframe = size(imgs, 3)
+    pp = fill(zero(M.x0), nframe, length(X0))
+
+    for k in 1:length(X0)
+        x = X0[k]
+        P = P0[k]
+
+        for t in 1:nframe
+            x, Ppred, A = Kalman.predict!(t, x, P, t + 1, M)
```

Comment:
Here you are `predict!`ing on frame # 1, how?

Reply:
but it seems like the initial guesses are not going to function so well as a "previous" estimate to where the root was one frame ago, wouldn't it be better to start predicting on frame #2?

Reply:
This follows their trajectories assuming the current position and speed.

Reply:
but how can we get an estimate for their speed from frame 1? Oh wait, we need to manually give the estimate, right?